### PR TITLE
[CephAdm][Tier1] Add test case for check-host and collocated rgw

### PR DESCRIPTION
**Pull request by MohitBis**
[CephAdm][Tier1] Add test case for check-host and collocated rgw
Check-host/prepare-host for cephadm use/host configuration
Negative Test case: Deploy collocated RGW with just node count

### CODE REVIEW

**Comment by MohitBis** [comment]
Logs:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-UP5Z6S
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-CQZD9W

File: tests/cephadm/test_collocate_rgw_node_count.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+def run(ceph_cluster, **kw):
+    """Deploy collocated RGW with just node count
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    config = kw.get("config")
+    node = ceph_cluster.get_nodes(role="mon")[0]
+    conf = {}
```
Comment (vamahaja) [inline tests/cephadm/test_collocate_rgw_node_count.py:19]: Change this to -
```
conf = { "pos_args": config.get("pos_args"), "placement": "2,count-per-host:2", }
```

File: tests/cephadm/test_collocate_rgw_node_count.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+from ceph.ceph import CommandFailed
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class Collocate_RGW_Node_Count(Exception):
```
Comment (vamahaja) [inline tests/cephadm/test_collocate_rgw_node_count.py:8]: Change class name to `CollocateRGWNodeCountError`

File: tests/cephadm/test_collocate_rgw_node_count.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    pass
+
+
+def run(ceph_cluster, **kw):
+    """Deploy collocated RGW with just node count
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
```
Comment (vamahaja) [inline tests/cephadm/test_collocate_rgw_node_count.py:15]: Add details which variables are expected in `**kw`

File: tests/cephadm/test_prepare_host.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+
+class PrepareHost(Exception):
+    pass
+
+
+def run(ceph_cluster, **kw):
+    """Verify prepare host command
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
```
Comment (vamahaja) [inline tests/cephadm/test_prepare_host.py:13]: Same as above

File: tests/cephadm/test_prepare_host.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class PrepareHost(Exception):
```
Comment (vamahaja) [inline tests/cephadm/test_prepare_host.py:6]: Change name to `PrepareHotsError`

File: tests/cephadm/test_prepare_host.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    """Verify prepare host command
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    config = kw.get("config")
+    node = ceph_cluster.get_nodes(role="mon")[0]
+
+    # run prepare host command
+    log.info("Running cephadm prepare-host command")
+    cmd = "cephadm prepare-host"
```
Comment (vamahaja) [inline tests/cephadm/test_prepare_host.py:20]: Instead add this command in `cli/cephadm/cephadm.py` library

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/cephadm/test_collocate_rgw_node_count.py (hunk 1 above)
Comment (MohitBis) [reply tests/cephadm/test_collocate_rgw_node_count.py]: Thanks @vamahaja 
Updated

File: tests/cephadm/test_collocate_rgw_node_count.py (hunk 2 above)
Comment (MohitBis) [reply tests/cephadm/test_collocate_rgw_node_count.py]: Thanks @vamahaja
Updated

File: tests/cephadm/test_collocate_rgw_node_count.py (hunk 3 above)
Comment (MohitBis) [reply tests/cephadm/test_collocate_rgw_node_count.py]: Thanks @vamahaja
Updated

File: tests/cephadm/test_prepare_host.py (hunk 4 above)
Comment (MohitBis) [reply tests/cephadm/test_prepare_host.py]: Thanks @vamahaja
Updated

File: tests/cephadm/test_prepare_host.py (hunk 5 above)
Comment (MohitBis) [reply tests/cephadm/test_prepare_host.py]: Thanks @vamahaja
Updated

File: tests/cephadm/test_prepare_host.py (hunk 6 above)
Comment (MohitBis) [reply tests/cephadm/test_prepare_host.py]: Thanks @vamahaja
Updated

**Comment by MohitBis** [comment]
Latest Logs:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-E6DZH5/
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-A73THF/

File: suites/pacific/cephadm/tier-1_service_apply_spec.yaml [hunk 7]
```diff
@@ -295,3 +295,12 @@ tests:
       desc: Adding admin node again with different labels  does not remove ceph.client.admin.keyring  and ceph.conf file
       polarion-id: CEPH-83574671
       module: test_admin_add_labels.py
+  - test:
+      name: test_deploy_collocated_RGW_with_node_count
+      desc: Deploy collocated RGW with just node count
+      config:
+        pos_args:
+          - foo
```
Comment (vamahaja) [inline suites/pacific/cephadm/tier-1_service_apply_spec.yaml:303]: `foo` ?? dummy variable ?

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/cephadm/test_collocate_rgw_node_count.py [hunk 8]
```diff
@@ -0,0 +1,41 @@
+from ceph.ceph import CommandFailed
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
```
[the remark this answers is not in the document]
Comment (MohitBis) [reply tests/cephadm/test_collocate_rgw_node_count.py]: Thanks @vamahaja 
Updated

File: tests/cephadm/test_prepare_host.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+                polarion-id: CEPH-83573789
+                module: test_prepare_host.py
+                config:
+                    result: Host looks OK
+    """
+    config = kw.get("config")
+    node = ceph_cluster.get_nodes(role="mon")[0]
+
+    # run prepare host command
+    log.info("Running cephadm prepare-host command")
```
[the remark this answers is not in the document]
Comment (MohitBis) [reply tests/cephadm/test_prepare_host.py:27]: Thanks @vamahaja
Updated

File: suites/pacific/cephadm/tier-1_service_apply_spec.yaml (hunk 7 above)
Comment (MohitBis) [reply suites/pacific/cephadm/tier-1_service_apply_spec.yaml:303]: @vamahaja 
It is service name as per the TC (Negative Scenario)

**Comment by MohitBis** [comment]
Latest Logs:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-4UTWM2/
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-5B5OJ2/

File: suites/pacific/cephadm/tier-1_service_apply_spec.yaml (hunk 7 above)
Comment (vamahaja) [reply suites/pacific/cephadm/tier-1_service_apply_spec.yaml:303]: ok

File: cli/cephadm/cephadm.py [hunk 10]
```diff
@@ -40,3 +41,15 @@ def rm_cluster(self, fsid, zap_osds=True, force=True):
...
+
+    def prepare_host(self, **kw):
+        """Verify prepare host command
+        Args:
+            kw (dict): Key/value pairs that needs to be provided to the installer.
+
+            Supported keys:
+                expect-hostname (str): hostname
+        """
+        cmd = f"cephadm prepare-host {build_cmd_from_args(**kw)}"
```
Comment (vamahaja) [inline cli/cephadm/cephadm.py:53]: This should be -
```
{self.base_shell_cmd} prepare-host {build_cmd_from_args(**kw)}
```

File: tests/cephadm/test_prepare_host.py [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+                name: test_prepare-host_command
+                desc: Check-host/prepare-host for cephadm use/host configuration
+                polarion-id: CEPH-83573789
+                module: test_prepare_host.py
+                config:
+                    result: Host looks OK
+    """
+    config = kw.get("config")
+    node = ceph_cluster.get_nodes(role="mon")[0]
+    result = CephAdm(node).prepare_host()
```
Comment (vamahaja) [inline tests/cephadm/test_prepare_host.py:23]: prepare_host [1] function require parameters, missing to pass ?
[1] https://github.com/red-hat-storage/cephci/pull/2167/files#diff-b8cd2f30f4dd9fabfedeb80881699702a81220bb949f17a3f1105fddb49975a0R45

File: cli/cephadm/cephadm.py [hunk 12]
```diff
@@ -40,3 +41,15 @@ def rm_cluster(self, fsid, zap_osds=True, force=True):
             cmd += " --force"
 
         return self.execute(sudo=True, long_running=True, cmd=cmd)
+
+    def prepare_host(self, **kw):
```
Comment (vamahaja) [inline cli/cephadm/cephadm.py:45]: If prepare-host command expects only one parameter then no need to use **kw, use `hostname` param

File: cli/cephadm/cephadm.py (hunk 10 above)
Comment (MohitBis) [reply cli/cephadm/cephadm.py]: Thanks @vamahaja 
Updated

File: tests/cephadm/test_prepare_host.py (hunk 11 above)
Comment (MohitBis) [reply tests/cephadm/test_prepare_host.py:23]: @vamahaja 

As per the test case we need to run command without any parameter.

File: cli/cephadm/cephadm.py (hunk 12 above)
Comment (MohitBis) [reply cli/cephadm/cephadm.py:45]: @vamahaja 
Since it is a optional parameter we have added **kw, Present test case not required hostname

**Comment by MohitBis** [comment]
Latest Log:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-82G1MV/

File: tests/cephadm/test_prepare_host.py (hunk 11 above)
Comment (vamahaja) [reply tests/cephadm/test_prepare_host.py:23]: ok

File: cli/cephadm/cephadm.py (hunk 12 above)
Comment (vamahaja) [reply cli/cephadm/cephadm.py:45]: ok

**Review by vamahaja — APPROVED** [review]

**Pull request merged**